File: learn-webflux/src/main/java/com/greatdreams/learn/webmvc/controller/HomeController.java
package com.greatdreams.learn.webmvc.controller;

import org.reactivestreams.Publisher;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;
import reactor.core.publisher.Mono;

@Controller
public class HomeController {
    @RequestMapping("/")
    @ResponseBody
    public Publisher<String> home() {
        return Mono.just("Hello , webflux");
    }
    @RequestMapping("/help.html")
    @ResponseBody
    public Publisher<String> help() {
        return Mono.just("Hello , webflux, ask for a help");
    }

    @RequestMapping("/login.html")
    @ResponseBody
    public String login() {
        return "login page";
    }
}

File: learn-webmvc/src/main/kotlin/com/greatdreams/learn/webmvc/controller/home_controller.kt
package com.greatdreams.learn.webmvc.controller

import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.ResponseBody


@Controller
class HomeController {
    @RequestMapping("/")
    @ResponseBody
    fun home(): String {
        return "Hello , Spring Webmvc"
    }

    @RequestMapping("/help.html")
    @ResponseBody
    fun help(): String {
        return "Hello , Spring Webmvc, ask for a help"
    }
}
